src/NHibernate/Mapping/SimpleValue.cs

```diff
@@ -324,7 +324,7 @@ public bool IsSimpleValue
 
 		public virtual bool IsValid(IMapping mapping)
 		{
-			return ColumnSpan == Type.GetColumnSpan(mapping);
+			return ColumnSpan == Type.GetOwnerColumnSpan(mapping);
```

Comment:
This will break something called "SpecialOneToOneType".

Reply:
Which seems to be unsupported anyway.

Reply:
Yeah. For the code below `ColumnSpan` is always 0. So it's never used. 
https://github.com/nhibernate/nhibernate-core/blob/c8b9ae6e7b4b839b21b22671e47adbba0707f0ad/src/NHibernate/Mapping/OneToOne.cs#L91-L103

Though I restored original behavior just in case...

Reply:
I think we should override logic for onetone.

Reply:
I've added `GetOwnerColumnSpan` override to `SpecialOneToOne` in my latest commit. Or did you mean something else?

Reply:
I was thinking to implement override for `IsValid` in `OneToOne` to handle this logic.